File: generated-sources/javascript-closure-angular/mojang-authentication/API/Client/GameProfile.js
goog.provide('API.Client.GameProfile');

/**
 * @record
 */
API.Client.GameProfile = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.GameProfile.prototype.agent;

/**
 * @type {!string}
 * @export
 */
API.Client.GameProfile.prototype.id;

/**
 * @type {!string}
 * @export
 */
API.Client.GameProfile.prototype.name;

/**
 * @type {!Object}
 * @export
 */
API.Client.GameProfile.prototype.userId;

/**
 * Unix timestamp in milliseconds
 * @type {!number}
 * @export
 */
API.Client.GameProfile.prototype.createdAt;

/**
 * @type {!boolean}
 * @export
 */
API.Client.GameProfile.prototype.legacyProfile;

/**
 * @type {!boolean}
 * @export
 */
API.Client.GameProfile.prototype.suspended;

/**
 * @type {!boolean}
 * @export
 */
API.Client.GameProfile.prototype.paid;

/**
 * @type {!boolean}
 * @export
 */
API.Client.GameProfile.prototype.migrated;

/**
 * @type {!boolean}
 * @export
 */
API.Client.GameProfile.prototype.legacy;

File: generated-sources/scalaz/mojang-authentication/src/main/scala/org/openapitools/client/api/GameProfile.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import GameProfile._

case class GameProfile (
  agent: Option[String],
id: Option[String],
name: Option[String],
userId: Option[File],
/* Unix timestamp in milliseconds */
  createdAt: Option[Long],
legacyProfile: Option[Boolean],
suspended: Option[Boolean],
paid: Option[Boolean],
migrated: Option[Boolean],
legacy: Option[Boolean])

object GameProfile {
  import DateTimeCodecs._

  implicit val GameProfileCodecJson: CodecJson[GameProfile] = CodecJson.derive[GameProfile]
  implicit val GameProfileDecoder: EntityDecoder[GameProfile] = jsonOf[GameProfile]
  implicit val GameProfileEncoder: EntityEncoder[GameProfile] = jsonEncoderOf[GameProfile]
}
